examples/routing_service/routing_service_file_adapter/c/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+#
+# (c) 2021 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+#  RTI grants Licensee a license to use, modify, compile, and create derivative
+#  works of the Software.  Licensee has the right to distribute object form
+#  only for use with RTI products.  The Software is provided "as is", with no
+#  warranty of any type, including any warranty for fitness for any purpose.
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+project(FileAdapterC)
+
+if(NOT DEFINED CONNEXTDDS_DIR)
+    set(msg 
+        "Please set the CMake variable CONNEXTDDS_DIR "
+        "to point to your installation of RTI Connext DDS")
+    message(FATAL_ERROR ${msg})
+endif()
+
+if (NOT UNIX)
+    set(msg 
+        "This example uses the pthreads library "
+        "which is only available on UNIX-like systems")
+    message(FATAL_ERROR ${msg})
+endif()
+
+# Find RTI Connext dependencies
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CONNEXTDDS_DIR}/resource/cmake")
+find_package(
+    RTIConnextDDS "6.0.0"
+    REQUIRED
+    COMPONENTS
+        core
+        routing_service
+)
+
+add_library(
+    fileadapter
+        "${CMAKE_CURRENT_SOURCE_DIR}/directory_reading.c"
+        "${CMAKE_CURRENT_SOURCE_DIR}/line_conversion.c"
+        "${CMAKE_CURRENT_SOURCE_DIR}/file_adapter.c"
+)
+
+target_link_libraries(
+    fileadapter
+        RTIConnextDDS::routing_service_infrastructure
+        ${CONNEXTDDS_EXTERNAL_LIBS}
+)
+
+# To differentiate between debug and release builds
+set_target_properties(fileadapter
+    PROPERTIES
+        OUTPUT_NAME_DEBUG fileadapterd
```

Comment:
This only work for multiconfiguration generators. Do you want to apply something similar to `Makefiles`?

Reply:
I didn't understand what you meant by `Makefiles`. I added this line since, if I compile we `-DCMAKE_BUILD_TYPE=Debug`, I want the library to have the right suffix. Failing to do that the release and debug mode libraries have the same name and overwrite each other.

Reply:
That is the reason why I'm asking:
`OUTPUT_NAME_<CONFIG>` only works (I think) for multi-configuration generators.

Anyway, since you just want to add a postfix, it is a better idea to use the `DEBUG_POSTFIX` CMake target property (https://cmake.org/cmake/help/latest/prop_tgt/DEBUG_POSTFIX.html)

Reply:
Done.